packages/base-types/src/models/version/prototype.ts

```diff
@@ -40,11 +40,24 @@ export interface PrototypeSettings {
   variableStateID?: string;
 }
 
-export interface Prototype<Command extends BaseCommand = BaseCommand, Locale extends string = string> {
+export type SurveyContext<SurveyContextExtension extends AnyRecord = AnyRecord> = {
+  slotsMap: Record<string, Slot>;
+  extraSlots: Slot[];
+  extraIntents: Intent[];
+  usedIntentsSet: string[];
+  platform: string; // VoiceflowConstants.PlatformType
+} & SurveyContextExtension;
+
+export interface Prototype<
+  Command extends BaseCommand = BaseCommand,
+  Locale extends string = string,
+  SurveyContextExtension extends AnyRecord = AnyRecord
+> {
   type: string;
   data: PrototypeData<Locale>;
   model: PrototypeModel;
   context: PrototypeContext<Command>;
   platform: string;
   settings: PrototypeSettings;
+  surveyorContext: SurveyContext<SurveyContextExtension>;
```

Comment:
This property is needed inside `general-service`. The `general-service` generates a surveyor context only when the prototype is rendered. May lead to less brittle code if we access the surveyor context through the `.prototype` property after it is rendered.

Otherwise, trying to return the surveyor context without accessing it through a finished `.protoype` (and thus invalidating the need for this type) can lead to more brittle code, in case the prototype rendering performs additional computation on the surveyor context. Then logic has to be duplicated to make sure the `surveyorContext` stored in a rendered prototype and the `surveyorContext` returned by the HTTP compile call match up.